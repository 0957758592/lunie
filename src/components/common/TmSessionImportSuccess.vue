<template>
  <SessionFrame :hide-back="true">
    <h2 class="session-title">
      Success!
    </h2>
    <div class="wrapper">
      <div class="content-main">
        <p>Your account has been created</p>
        <TmBtn value="Back to Lunie" @click.native="$router.push('/')" />
      </div>
    </div>
  </SessionFrame>
</template>

<script>
import TmBtn from "common/TmBtn"
import SessionFrame from "common/SessionFrame"

export default {
  name: `session-session-success`,
  components: {
    TmBtn,
    SessionFrame
  },
  created() {
    this.$store.commit("recoverReset")
  }
}
</script>

<style scoped>
.wrapper {
  position: relative;
  width: var(--width-main);
  display: flex;
  height: 100%;
  margin: 1rem auto;
}

.content-main {
  display: flex;
  flex-flow: column;
}

p {
  margin: 1rem auto;
}
</style>
