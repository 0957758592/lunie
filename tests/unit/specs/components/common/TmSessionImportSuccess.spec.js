import Vuex from "vuex"
import Vuelidate from "vuelidate"
import { shallowMount, createLocalVue } from "@vue/test-utils"
import TmSessionImportSuccess from "common/TmSessionImportSuccess"
// import TmBtn from "common/TmBtn"
jest.mock(`scripts/google-analytics.js`, () => () => {})
const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(Vuelidate)
localVue.directive(`tooltip`, () => {})
localVue.directive(`focus`, () => {})

describe(`TmSessionImportSuccess`, () => {
  let wrapper, $store, getters

  beforeEach(() => {
    getters = {}
    $store = {
      state: {
        recover: {
          name: ``,
          seed: ``,
          password: ``,
          passwordConfirm: ``,
          warning: false
        }
      },
      getters,
      commit: jest.fn(),
      dispatch: jest.fn(async () => true)
    }
    wrapper = shallowMount(TmSessionImportSuccess, {
      localVue,
      mocks: {
        $store,
        $router: {
          push: jest.fn()
        }
      },
      stubs: [`router-link`]
    })
  })

  it(`has the expected html structure`, async () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`should dispatch resetRecoverData`, async () => {
    expect($store.dispatch).toHaveBeenCalledWith(`resetRecoverData`)
  })
})
