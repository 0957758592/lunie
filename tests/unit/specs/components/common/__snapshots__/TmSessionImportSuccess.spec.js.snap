// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmSessionImportSuccess has the expected html structure 1`] = `
<sessionframe-stub
  hideback="true"
>
  <h2
    class="session-title"
  >
    
    Success!
  
  </h2>
   
  <div
    class="session-main"
  >
    <p>
      Your account has been created
    </p>
  </div>
   
  <div
    class="session-footer"
  >
    <tmbtn-stub
      value="Back to Lunie"
    />
  </div>
</sessionframe-stub>
`;
